swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh

```diff
@@ -33,4 +38,49 @@ function xccov_to_generic {
   echo '</coverage>'
 }
 
+function check_sdk_version {
+  sdk_major_version=`xcrun --show-sdk-version | cut -d . -f 1`
+
+  if [ $? -ne 0 ]; then 
+    echo 'Failed to execute xcrun show-sdk-version' 1>&2
+    exit -1
+  fi
+  echo $sdk_major_version
+}
+
+function cleanup_tmp_files {
+  rm -rf tmp.json
+  rm -rf tmp.xccovarchive
+}
+
+# Optimize coverage files conversion time by exporting to a clean xcodearchive directory
+# Credits to silverhammermba on issue #68 for the suggestion
+function optimize_format {
+  cleanup_tmp_files
+  xcrun xcresulttool get --format json --path "$xccovarchive_file" > tmp.json
+  if [ $? -ne 0 ]; then 
+    echo 'Failed to execute xcrun xcresulttool get' 1>&2
+    exit -1
+  fi
+
+  # local reference=$(jq -r '.actions._values[2].actionResult.coverage.archiveRef.id._value'  tmp.json)
+  local reference=$(jq -r '.actions._values[]|[.actionResult.coverage.archiveRef.id],._values'  tmp.json | grep value | cut -d : -f 2 | cut -d \" -f 2)
+  if [ $? -ne 0 ]; then 
+    echo 'Failed to execute jq (https://stedolan.github.io/jq/)' 1>&2
+    exit -1
+  fi
+  xcrun xcresulttool export --type directory --path "$xccovarchive_file" --id "$reference" --output-path tmp.xccovarchive
```

Comment:
@sylvain-combe-sonarsource , if the $reference here ends up being a list of IDs (from a merged .xcresult bundle of multiple test plans), can we iterate over them and have the same output-path?

Reply:
Something like this shows promise:
```
for test_ref in $reference; do
     xcrun xcresulttool export --type directory --path "$xccovarchive_file" --id "$test_ref" --output-path tmp.xccovarchive
done
```

Reply:
@mcstoufer
Many thanks for your review and this proposal. 
I was not able to test it fully as I don't have a project with multiple test plans or with merged .xcresult bundles.
But I've added your proposed change to the PR, please confirm it works for you and I'll merge it.